src/power/power.c

```diff
@@ -0,0 +1,1401 @@
+         * default values.
+         */
+
+        memset(&baryonData, 0, sizeof baryonData); // 0x00000618
+
+        g_Power.wlanExclusivePllClockLimit = SCE_POWER_WLAN_EXCLUSIVE_PLL_CLOCK_LIMIT_222Mhz; // 0x00000624
+        g_Power.watchDog = 0; // 0x00000630
+        g_Power.isWlanSuppressChargingEnabled = SCE_FALSE; // 0x00000638
+        g_Power.backlightMaximumWlanActive = 0; // 0x0000063C
+        g_Power.ledOffTiming = SCE_POWER_LED_OFF_TIMING_AUTO; // 0x00000644
+
+        forceSuspendBatteryCapacity = BATTERY_FORCE_SUSPEND_CAPACITY_THRESHOLD; // 0x0000062C
+        lowBatteryCapacity = BATTERY_LOW_CAPACITY_THRESHOLD; // 0x00000634
+
+        /* 
+         * If the IdStorage is corrupted or the data could not be obtained for other reasons,
+         * the force suspend capacity is set to the low battery capacity threshold.
+         */
+        _scePowerChangeSuspendCap(BATTERY_LOW_CAPACITY_THRESHOLD); // 0x00000640
+    }
+    else
+    {
+        /* Store the successfully obtained BARYON ID storage configuration data. */
+
+        g_Power.ledOffTiming = baryonData.ledOffTiming; // 0x0000006C & 0x00000080
+        g_Power.watchDog = baryonData.watchDog & 0x7F; // 0x00000088
+        g_Power.isWlanSuppressChargingEnabled = baryonData.isWlanSuppressChargingEnabled; // 0x00000070 & 0x0000008C
+        g_Power.backlightMaximumWlanActive = baryonData.backlightMaximumWlanActive; // 0x00000074 & 0x00000090
+
+        forceSuspendBatteryCapacity = baryonData.forceSuspendBatteryCapacity; // 0x00000094
+        lowBatteryCapacity = baryonData.lowBatteryCapacity; // 0x0000009C
+
+        g_Power.wlanExclusivePllClockLimit = (baryonData.wlanExclusivePllClockLimit >= 0x80) // 0x00000064 & 0x0000007C & 0x00000098
+            ? baryonData.wlanExclusivePllClockLimit & 0x7F // 0x00000084 & 0x000000A4
+            : 1; // 0x000000A0 & 0x000000A4
+    }
+
+    s32 appType = sceKernelApplicationType(); // 0x000000A8
+    if (appType == SCE_INIT_APPLICATION_GAME) // 0x000000B4
+    {
+        /* Initially limit WLAN's PLL coexistence clock frequency to 222MHz when we are a game application. */
+        g_Power.wlanExclusivePllClockLimit = SCE_POWER_WLAN_EXCLUSIVE_PLL_CLOCK_LIMIT_222Mhz; // 0x000005FC
+
+        /* 
+         * For game applications, the PLL clock frequency can only be set to the following values (in MHz):
+         * 199, 222, 266 and 333.
+         */
+        u32 pllUseMask = SCE_POWER_PLL_USE_MASK_190MHz | SCE_POWER_PLL_USE_MASK_222MHz
+            | SCE_POWER_PLL_USE_MASK_266MHz | SCE_POWER_PLL_USE_MASK_333MHz;
+        scePowerSetPllUseMask(pllUseMask); // 0x00000600
+    }
+
+    /* Determine whether USB charging is supported by the PSP device we are running on. */
+
+    u8 baryonVersionMajor = PSP_SYSCON_BARYON_GET_VERSION_MAJOR(g_Power.baryonVersion);
+    u8 baryonVersionMinor = PSP_SYSCON_BARYON_GET_VERSION_MINOR(g_Power.baryonVersion);
+    if (baryonVersionMajor == 0x2 

```

Comment:
This could be greatly simplified, but it seems like we're trying to match Sony's implementation which must have had the optimizer turned off or something.